src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -102,9 +107,18 @@ public IEnumerable<string> GetManifestDirectories()
 
                 foreach (var workloadManifestDirectory in directoriesWithManifests.Values)
                 {
-                    yield return workloadManifestDirectory;
+                    if (!IsManifestIdOutdated(workloadManifestDirectory))
+                    {
+                        yield return workloadManifestDirectory;
+                    }
                 }
             }
         }
+
+        private bool IsManifestIdOutdated(string workloadManifestDir)
+        {
+            var manifestId = Path.GetFileName(workloadManifestDir).ToLowerInvariant();
```

Comment:
If `_outdatedManifestIds` would be created with `StringComparer.OrdinalIgnoreCase`, this would no longer be necessary:
```suggestion
            var manifestId = Path.GetFileName(workloadManifestDir);
```